docs/DeveloperGuide.md

```diff
@@ -492,3 +501,68 @@ testers are expected to do more *exploratory* testing.
+
+4. **Allow more flexibility in command prefixes:**
+   * Currently, the command prefixes are case-sensitive.
+     * For example, `n/` is the prefix for "name" parameter, but `N/` is not acceptable by the command parser.
+   * It may provide more convenience to user by disabling case sensitivity for prefixes. For example, allowing `n/` and `N/` as the prefix for "name" parameter.
+
+5. **Supporting more forms of name:**
+   * Currently, only name form that consist of alphanumeric characters and spaces in between is allowed.
+   * It may improve the diversity of the application, by supporting name in different forms, including names in different languages and names that contain non-alphanumeric symbols.
+
+6. **Supporting more forms of room number:**
+   * Currently, the room number format is {block}-{floor}-{room number}, where block and room number are at least 2 alphanumeric characters and floor is strictly 2 alphanumeric characters. E.g. `nw-12-12`.
+   * To accommodate more room number format, it will be good to change the minimum length of block and room number as 1.
+
+7. **Allow more flexibility in free time tag format:**
+   * Currently, free time tag only accept forms of `DDD:HHmm-HHmm`, e.g. `Mon:1200-1300`.
+   * It may provide more convenience to user by disabling case sensitivity on the day part, e.g. allowing `mon:1000-1200`, `Mon:1000-1200`, `MON:1000-1200`.
+
+8. **Support more accurate free time:**
+   * A person may not always be free for the same time interval every week. For example, a person can be free at `Mon:1000-1200` for every odd week of the semester, and busy at the time for every even week of the semester.
+   * A possible solution is to specify actual date as part of the free time interval, e.g. `01/01:Mon:1000-1200`.
+     * Pros: It provides more customizable user experience.
+     * Cons: It requires users to have detailed schedule information of a person in their contact list, where the person may not be willing to disclose. It also requires users to provide more complicated parameters, which is more troublesome and error-prone.
+   * Another possible solution is to host the application online, and allow user to specify their own free times. 
+     * Pros: User may connect to the real-time state of another user, and able to view the real-time updated free time of others.
+     * Cons: More changes may need to be done on the application, and the developer team will need to re-evaluate the necessity of certain commands. 
+       * For example, users may not need to manually input and record the detail of others, as they only need to update their own profile, and others can view the real-time state of it. 
+       * Hence, some commands e.g. `addTime`, `deleteTime`, and `edit`, are no longer applicable to the application.
+     
+9. **Make error messages more specific:**
```

Comment:
We need to specify which error message we targeting I think, below is the example they provided:
![image](https://github.com/AY2324S2-CS2103T-F11-4/tp/assets/65356386/1eae122d-f53f-4c7e-99e9-09ffb29e3415)

Reply:
Will change in the next update.